server/src/test/java/org/elasticsearch/gateway/RecoverAfterNodesIT.java

```diff
@@ -40,6 +42,23 @@
 public class RecoverAfterNodesIT extends ESIntegTestCase {
     private static final TimeValue BLOCK_WAIT_TIMEOUT = TimeValue.timeValueSeconds(10);
 
+    @Override
+    protected List<Settings> updateNodesSettings(List<Settings> allNodesSettings) {
```

Comment:
This seems overly general here, since this test suite only ever starts one node at once. Can we just expose the first node's name (e.g. `InternalTestCluster#buildNodeName`) and set the setting directly in the places where, today, we set `ClusterBootstrapService.INITIAL_MASTER_NODE_COUNT_SETTING`?

Reply:
The thing is that `InternalTestCluster` does not allow to override `node.name` in node settings, so the test does not know the node name in advance, that's why we have to add bootstrap settings, once all settings including the `node.name` are available.

Reply:
Sure, what I meant was that `InternalTestCluster#buildNodeName` can tell us the name that the first node will have, so why not just call that and use it when starting up the first node?